QKSMS/src/main/java/com/moez/QKSMS/transaction/SmsHelper.java

```diff
@@ -542,6 +542,33 @@ public static int getUnreadMessageCount(Context context) {
         return messages;
     }
 
+    public static List<Message> deleteFailedMessages(Context context) {
```

Comment:
This deletes all failed messages, not the ones for a particular thread. If the dialog that allows you to do this comes from long-pressing on a conversation, then it should only delete the messages from that conversation